### PR TITLE
Add smoke test for PyroModule based linear regression

### PR DESCRIPTION
**Pull request by neerajprad**
I was exploring if we could use `PyroModule` to replace `random_module` but got into issues like:
 - Site names containing `.` and being rejected by `nn.Module`.
 - Site names not matching between guide/model traces.

This failing test captures the issues involved.

### CODE REVIEW

File: tests/infer/test_autoguide.py [hunk 1]
```diff
@@ -568,10 +568,53 @@ def forward(self):
...
+            self.weight = PyroSample(lambda self: dist.Normal(0., 1.).expand([out_features, in_features])
+                                     .to_event(2))
+            self.bias = PyroSample(lambda self: dist.Normal(0., 10.).expand([out_features]).to_event(1))
+
+    class LinearRegression(PyroModule):
+        def __init__(self):
+            super().__init__()
+            self.linear = RandomLinear(D, 1)
+
+        def forward(self, x, y):
```
Comment (neerajprad) [inline tests/infer/test_autoguide.py]: @fritzo - Does this look correct?

Comment (fritzo) [reply tests/infer/test_autoguide.py]: Looks right to me.

**Comment by martinjankowiak** [comment]
@neerajprad :+1: it is my belief that we should consider deprecating lift/random_module in favor of PyroModule. the former paradigm seems to cause a lot of confusion among users

**Comment by neerajprad** [comment]
The tests pass now. This is so much cleaner than `random_module`. 😆

**Comment by neerajprad** [comment]
Ready to merge.

**Pull request merged**